ts/src/main/scala/ch/epfl/ts/engine/rules/FxMarketRulesWrapper.scala

```diff
@@ -0,0 +1,90 @@
+package ch.epfl.ts.engine.rules
+
+import ch.epfl.ts.engine.{OrderBook, ForexMarketRules}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency.Currency
+import scala.collection.mutable
+import scala.Some
+import ch.epfl.ts.data.MarketBidOrder
+import ch.epfl.ts.data.LimitBidOrder
+import scala.Some
+import ch.epfl.ts.data.MarketAskOrder
+import ch.epfl.ts.data.LimitAskOrder
+
+/**
+ * Wrapper around ForexMarketRules, to have ALL the information needed to proceed the order in one method.
+ * Created by sygi on 09.05.15.
+ */
+class FxMarketRulesWrapper extends MarketRulesWrapper(new ForexMarketRules()) {
+  val rules = new ForexMarketRules()
```

Comment:
I see two `new ForexMarketRules`, couldn't you only use a `val` constructor parameter in the `MarketRulesWrapper`?